src/map/battle.c

```diff
@@ -4916,11 +4915,19 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 	}
 
 	//Check for critical
+#ifdef RENEWAL
+	if ((battle_config.feature_enable_multi_crit == 1 || !(wd.type&BDT_MULTIHIT)) && sstatus->cri &&
+		(!skill_id ||
```

Comment:
There is no need to copy the skill id's part, only ifdef the part that is actually different.

Reply:
Can't do the ifdef parts only cause those ifdef's are one line.

Reply:
Do you have a suggestion on how I can do that?

Reply:
It should be something like, but the usage of the `&` operator here is wrong anyways, `wd.type` is not a bitmask.

```C
	if (
#ifdef RENEWAL
		(battle_config.feature_enable_multi_crit == 1 || !(wd.type&BDT_MULTIHIT)) &&
#else
		!flag.cri && wd.type != BDT_MULTIHIT &&
#endif
		sstatus->cri &&
		(!skill_id ||
		skill_id == KN_AUTOCOUNTER ||
		skill_id == SN_SHARPSHOOTING || skill_id == MA_SHARPSHOOTING ||
		skill_id == NJ_KIRIKAGE))
```

Reply:
Done, please check the new commit 865838f